Implemented minor improvements (#4775)

reisfmb opened the pull request:
- Fixed file uploader;

- Removed '?' from generated content link

- Added media options visibility after file upload

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -800,11 +800,20 @@ export class LinkModalDialog
         mediaUploader.onFileUploaded((event: UploadedEvent<Content>) => {
-            let item = event.getUploadItem();
-            let createdContent = item.getModel();
+            const mediaRadio = <RadioGroup>this.getFieldById('contentMediaRadio');
```
I insist on us using static typing for all variables, even if the type can be inferred from looking on the var

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -800,11 +800,20 @@ export class LinkModalDialog
+                checkbox.show();
+            }
+
+            const item = event.getUploadItem();
```
Missing types

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -901,7 +913,7 @@ export class LinkModalDialog
         };
 
-        if (contentSelectorSelectedContent.getType().isDescendantOfMedia()) {
+        if (contentSelectorSelectedContent && contentSelectorSelectedContent.getType().isDescendantOfMedia()) {
```
`if (contentSelectorSelectedContent?.getType().isDescendantOfMedia()) {...`

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -888,10 +897,13 @@ export class LinkModalDialog
 
             const fragment = anchorString ? `${LinkModalDialog.fragmentPrefix}${anchorString}`: '';
```
`const fragment: string = ....`

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -868,7 +878,6 @@ export class LinkModalDialog
         const contentSelectorSelectedContent = contentSelector.getSelectedContent();
 
-
         const getQueryParamsString = () => {
```
extract all internal functions to private class methods

ashklianko commented on modules/lib/src/main/resources/assets/js/app/inputtype/ui/text/dialog/LinkModalDialog.ts:
```diff
@@ -888,10 +897,13 @@ export class LinkModalDialog
         };
 
-
         const setUrl = () => {
```
extract all internal functions to private class methods

setUrl name is confusing since you're not only setting url within it

ashklianko commented:
Hi Bruno!
In general `createContentLink` should be refactored:
-All internal functions should be private class functions
-All vars should have types set
-Functions should do just what they say they do, if this is `setUrl()` then it should only set the url :)

reisfmb committed:
Implemented minor improvements (#4775)

- Fixed file uploader;

- Removed '?' from generated content link

- Added media options visibility after file upload

- Refactored createContentLink

ashklianko committed:
small ref

reisfmb committed:
Merge pull request #4784 from enonic/issue-4775_small_ref

small ref

reisfmb committed:
Linting (#4775)

reisfmb committed:
Tests fix (#4775)

reisfmb committed:
XP 11 (#4775)

ashklianko committed:
small ref

ashklianko committed:
small ref

reisfmb committed:
Merge pull request #4785 from enonic/issue-4775_upd

small ref
